web/docs/language/features.md

```diff
@@ -443,6 +443,149 @@ import { isPrismaError, prismaErrorToHttpError } from '@wasp/utils.js'
+
+#### pg-boss
+
+We have selected [pg-boss](https://github.com/timgit/pg-boss/) as our first job executor to handle the low-volume, basic job queue workloads many web applications have (think sending emails, connecting to external HTTP APIs, etc.). By using PostgreSQL (and [SKIP LOCKED](https://www.2ndquadrant.com/en/blog/what-is-select-skip-locked-for-in-postgresql-9-5/)) as it's storage and synchronization mechanism, it allows us to provide many job queue pros without any additional infrastructure or complex management.
+
+### Basic job definition and usage
+
+To register a `job` in Wasp, simply add a declaration with a reference to an `async` function, like the following:
+
+```css title="main.wasp"
+job mySpecialJob {
+  executor: PgBoss,
+  perform: {
+    fn: import { foo } from "@ext/jobs/bar.js",
+    options: {=json { "retryLimit": 1 } json=} // optional
+  }
+}
+```
+
+Then, in your [Operations](/docs/language/features#queries-and-actions-aka-operations) or [setupFn](/docs/language/features#setupfn-extimport-optional), you can enqueue work to be done:
+```js
+import { mySpecialJob } from '@wasp/jobs/mySpecialJob.js'
+
+const submittedJob = await mySpecialJob.submit({ job: "args" })
+console.log(await submittedJob.pgBoss.details())
+
+// Or, if you'd prefer it to execute in the future, just add a .delay().
+// It takes a number of seconds, Date, or ISO date string.
+await mySpecialJob.delay(10).submit({ job: "args" })
+```
+
+And that is it! Your job will be executed by the job executor as if you called `foo({ job: "args" })`.
+
+### Recurring jobs
+
+If you have work that needs to be done on some recurring basis, you can add a `schedule` to your job declaration:
+
+```css  {6-10} title="main.wasp"
+job mySpecialJob {
+  executor: PgBoss,
+  perform: {
+    fn: import { foo } from "@ext/jobs/bar.js"
+  },
+  schedule: {
+    cron: "0 * * * *",
+    performFnArg: {=json { "job": "args" } json=}, // optional
+    options: {=json { "retryLimit": 2 } json=} // optional
+  }
+}
+```
+
+In this example, you do _not_ need to invoke anything in JavaScript. You can imagine `foo({ "job": "args" })` getting automatically scheduled and invoked for you every hour.
+
+### Fields
+
+#### `executor: JobExecutor` (required)
+The executor can be either `Passthrough` or `PgBoss`. `Passthrough` is a zero-dependency executor used mainly for testing purposes. `PgBoss` is the recommended executor for low-volume, production use cases, and requires your `app.db.system` to be `PostgreSQL`.
+
+####  `perform: dict` (required)
+
+  - ##### `fn: fn` (required)
+  An `async` JavaScript function of work to be performed. It can take any number of arguments.
+  
+  - ##### `options: JSON` (optional)
+  Executor specific default options to use when enquing jobs. It can be overriden during invocation with `submit()`.
+
+  > See the docs for [pg-boss](https://github.com/timgit/pg-boss/blob/master/docs/readme.md#sendname-data-options).
```

Comment:
Should we emphasize more that these options are directly passed to pg-boss, and that they are therefore pg-boss options and should be looked up in their docs? Also, are we careful to link the version of their docs that match the version of pg-boss we use in Wasp? Hm if we are that will not be very easy to update manually, we might want to make some smart mechanism for that in our blog, maybe centralize versions of stuff that we use.

Reply:
Yeah, I can be more explicit these are just passed straight in and should be looked up there. We could link to the specific version we use (although we do use ^ in the sem ver but should be ok). Yeah the accidental drift if we forget to update it may be a problem though 🤔